tests/platform_tests/api/test_fan.py

```diff
@@ -0,0 +1,208 @@
+            except:
+                pytest.fail("num_fans is not an integer")
+    
+    #
+    # Functions to test methods inherited from DeviceBase class
+    #
+    def test_get_name(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_fans):
+            name = fan.get_name(platform_api_conn, i)
+            if self.expect(name is not None, "Unable to retrieve Fan {} name".format(i)):
+                self.expect(isinstance(name, str), "Fan {} name appears incorrect".format(i))
+        self.assert_expectations()
+
+    def test_get_presence(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_fans):
+            presence = fan.get_presence(platform_api_conn, i)
+            if self.expect(presence is not None, "Unable to retrieve fan {} presence".format(i)):
+                if self.expect(isinstance(presence, bool), "Fan {} presence appears incorrect".format(i)):
+                    self.expect(presence is True, "Fan {} is not present".format(i))
+        self.assert_expectations()
+
+    def test_get_model(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_fans):
+            model = fan.get_model(platform_api_conn, i)
+            if self.expect(model is not None, "Unable to retrieve fan {} model".format(i)):
+                self.expect(isinstance(model, str), "Fan {} model appears incorrect".format(i))
+        self.assert_expectations()
+
+    def test_get_serial(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_fans):
+            serial = fan.get_serial(platform_api_conn, i)
+            if self.expect(serial is not None, "Unable to retrieve fan {} serial number".format(i)):
+                self.expect(isinstance(serial, str), "Fan {} serial number appears incorrect".format(i))
+        self.assert_expectations()
+
+    def test_get_status(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_fans):
+            status = fan.get_status(platform_api_conn, i)
+            if self.expect(status is not None, "Unable to retrieve fan {} status".format(i)):
+                self.expect(isinstance(status, bool), "Fan {} status appears incorrect".format(i))
+        self.assert_expectations()
+
+    #
+    # Functions to test methods defined in FanBase class
+    #
+
+    def test_get_speed(self, duthost, localhost, platform_api_conn):
+        # Ensure the fan speed is sane
+        for i in range(self.num_fans):
+            speed = fan.get_speed(platform_api_conn, i)
+            logger.error("vaibhav got fan speed {}".format(speed))
+            if self.expect(speed is not None, "Unable to retrieve Fan {} speed".format(i)):
+                if self.expect(isinstance(speed, int), "Fan {} speed appears incorrect".format(i)):
+                    self.expect(speed > 0 and speed <= 100, "Fan {} speed  {} reading is not within range".format(i, speed))
```

Comment:
Delete extra space between `speed` and `{}`

Reply:
fixed